Sources/MapboxNavigation/NavigationMapView.swift

```diff
@@ -102,13 +98,9 @@ open class NavigationMapView: UIView {
     var currentRestrictedAreasStops = [Double: UIColor]()
     var routeLineTracksTraversal: Bool = false {
         didSet {
-            if routeLineTracksTraversal, let route = self.routes?.first {
-                initPrimaryRoutePoints(route: route)
-                setUpLineGradientStops(along: route)
-            } else {
-                removeLineGradientStops()
+            if let route = self.routes?.first {
+                show([route], legIndex: currentLegIndex)
             }
-            updateRestrictedAreasGradientStops(along: self.routes?.first)
```

Comment:
A side effect of removing this line: if there are no `routes` available, previous logic would clear existing `currentRestrictedAreasStops` while new implementation does not. It shouldn't be visible or introduce visual glitches, but would be nice to do to maintain view state.

Reply:
Thanks for the suggestion, I'll add the `updateRestrictedAreasGradientStops(along:)` when no `routes` available to empty the `currentRestrictedAreasStops`.